bitcoin/BRSyncManager.c

```diff
@@ -0,0 +1,1549 @@
+//  BRSyncManager.h
+//
+//  Created by Michael Carrara on 12/08/19.
+//  Copyright (c) 2019 breadwallet LLC.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+#include <inttypes.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <pthread.h>
+
+#include "support/BRArray.h"
+#include "BRSyncManager.h"
+#include "BRPeerManager.h"
+
+/// MARK: - Common Decls & Defs
+
+#if !defined (MAX)
+#define MAX(a,b) (((a)>(b))?(a):(b))
+#endif
+
+#if !defined (MIN)
+#define MIN(a,b) (((a)<(b))?(a):(b))
+#endif
+
+struct BRSyncManagerStruct {
+    BRSyncMode mode;
+};
```

Comment:
Certain other common things can already be moved into here?  `lock` and `isConnected` at least?

Reply:
Came down to preference. I can move but has no practical impact. I personally like looking at the struct and seeing all its fields in one place rather than bouncing around.